### PR TITLE
Bindings

### PR DESCRIPTION
**Pull request by Anderssorby**
A simple proof of concept binding which hashes a single Scalar input as a ByteArray LE.

### CODE REVIEW

File: src/Neptune.lean [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
+BLS12-381 pairing-friendly elliptic curve construction
+-/
+namespace Blstrs
+constant SCALAR_BYTES_LE: Nat := 32
 
 /-
 A dependent ByteArray which guarantees the correct byte length.
 -/
-def NeptuneHash : Type := { r : ByteArray // r.size = NEPTUNE_OUT_LEN }
+def Scalar : Type := { r : ByteArray // r.size = SCALAR_BYTES_LE }
```
Comment (johnchandlerburnham) [inline src/Neptune.lean:20]: Scalar is a field element? Should be mentioned in the doc comment if so

File: src/Neptune.lean [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
 def Hasher : Type := HasherPointed.type
 
 instance : Inhabited Hasher := ⟨HasherPointed.val⟩
 
+@[extern "lean_neptune_poseidon"]
+private constant lean_neptune_poseidon (bs : @& ByteArray) : ByteArray
 
-def hash {I: Type u} [Into ByteArray I] (input : I) : NeptuneHash :=
-  Inhabited.default
+def poseidon {I: Type u} [Into Blstrs.Scalar I] (input : I) : ByteArray :=
```
Comment (johnchandlerburnham) [inline src/Neptune.lean]: Should this return `Scalar`? Or do we not have the size constraint info when going through the extern?

File: src/Neptune/BinaryTools.lean [hunk 3]
```diff
@@ -34,12 +34,17 @@ export Into (into)
...
+instance (A B C: Type u) [Into B C] [Into A B] : Into A C := ⟨fun c : C =>
+          let b: B := Into.into c;
+          Into.into b⟩
 
 instance : Into ByteArray String := {
-  into := String.toByteArray
+  into := String.toUTF8
 }
 
 namespace Alphabet
```
Comment (johnchandlerburnham) [inline src/Neptune/BinaryTools.lean:50]: This is from an older multibase implementation, can we import the updated one?

**Comment by johnchandlerburnham** [comment]
We also should get CI passing

File: src/Neptune.lean (hunk 1 above)
Comment (Anderssorby) [reply src/Neptune.lean:20]: Yes, but I don't understand it very well yet so I figured I would document it better once I know what it is used for.

File: src/Neptune/BinaryTools.lean (hunk 3 above)
Comment (Anderssorby) [reply src/Neptune/BinaryTools.lean:50]: The problem is that Multibase currently imports Neptune. Maybe split Multibase into it's own repo?

**Comment by porcuquine** [comment]
Only had time to glance so far, but a quick note which may matter: you can’t hash full 32-byte chunks as one field element of the preimage. The largest size guaranteed to fit in a field element (BLS12-381) is 254 bits. This might not be relevant, but I thought it might be.

**Comment by Anderssorby** [comment]
@porcuquine I'm really just matching the type signature of the rust function.

**Pull request merged**